### PR TITLE
Patrick/vtu support

### PR DESCRIPTION
**Pull request by destravous**
adds support for loading vtu files with polyhedron data

### CODE REVIEW

File: src/xml.rs [hunk 1]
```diff
@@ -1611,6 +1684,98 @@ impl DataArray {
...
+            DataArrayFormat::Appended => {
+                return Err(ValidationError::Unsupported);
+            }
+            DataArrayFormat::Binary => {
+                let header_bytes = ei.header_type.size();
+                use model::IOBuffer;
+                if matches!(ei.compressor, Compressor::None) {
+                    // First byte gives the bytes
+                    let bytes =
+                        BASE64_STANDARD.decode(data.into_iter().next().unwrap().into_string())?;
```
Comment (bitcooker) [inline src/xml.rs:1711]: Maybe change this `unwrap` to a more descriptive `expect`

**Review by bitcookr — CHANGES_REQUESTED** [review]

File: src/xml.rs (hunk 1 above)
Comment (bitcooker) [reply src/xml.rs:1711]: As well as the one on line 1757

File: src/xml.rs [hunk 2]
```diff
@@ -599,30 +600,55 @@ mod topo {
         where
             A: MapAccess<'de>,
         {
-            let make_err = || {
-                <A::Error as serde::de::Error>::custom(
-                "Cells data arrays must contain three DataArrays named \"connectivity\", \"offsets\" and \"types\""
-            )
-            };
+            let error_str = format!("Cells data arrays must contain at least three DataArrays named \"connectivity\", \"offsets\" and \"types\";");
```
Comment (SarthakSingh31) [inline src/xml.rs]: Minor nit. This format is unnecessary. It can just be a `&str`.

File: src/xml.rs (hunk 1 above)
Comment (SarthakSingh31) [reply src/xml.rs]: Here I think the `into_string` that returns a string should be changed to a `as_str` method that returns a `&str`. Then you can do `data[0].as_str()` which is much more idomatic

**Review by SarthakSingh31 — APPROVED** [review]

**Pull request merged**